### PR TITLE
Task/video 10785/improvement remote storage to parse error using retrofit

### PR DESCRIPTION
**Pull request by alejo740**
### JIRA link(s):
- [VIDEO-10785](https://issues.corp.twilio.com/browse/VIDEO-10785)

### Description

- Refactor for `RemoteStorage` process to validate and parsing Error responses using the internal retrofit parsing process.

After a while researching a looking for alternatives to move the error parsing process to a common place, we have decided the best way for now is using a generic method for all methods in the RemoteStorage class.

However, I have found a new way to make the parsing process better using the original `retrofit` instance.

### CODE REVIEW

**Review by timrozum — COMMENTED** [review]
I'll defer to @afalls-twilio

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt [hunk 1]
```diff
@@ -1,53 +1,61 @@
...
+    private inline fun <reified T : BaseResponse> validateResponse(
         response: Response<T>,
     ): T {
-        val result: T = response.body() ?: T::class.java.newInstance()
+        val result: T = if (!response.isSuccessful) { // Failure case
+            response.errorBody()?.let { body ->
+                try {
+                    retrofit.responseBodyConverter<T>(T::class.java, arrayOf()).convert(body)
+                } catch (e: MalformedJsonException) {
+                    null
```
Comment (afalls-twilio) [inline apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt:44]: this error should be trapped some how and at least logged to logcat

**Review by afalls-twilio — CHANGES_REQUESTED** [review]

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt (hunk 1 above)
Comment (alejo740) [reply apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt:55]: Done!

**Review by afalls-twilio — APPROVED** [review]
LGTM

**Pull request merged**